Alpaca.Markets/RestClient.cs

```diff
@@ -64,16 +81,28 @@ public sealed partial class RestClient : IDisposable
         /// <param name="polygonRestApi">Polygon REST API ennpoint URL.</param>
         /// <param name="alpacaDataApi">Alpaca REST data API endpoint URL.</param>
         /// <param name="isStagingEnvironment">If <c>true</c> use staging.</param>
+        /// <param name="maxRetryAttempts">Number of times to retry an Http request, if the status code is one of the <paramref name="retryHttpStatuses"/></param>
+        /// <param name="retryHttpStatuses">Http status codes that trigger a retry, up to the <paramref name="maxRetryAttempts"/></param>
+        /// <param name="apiVersion">Version of Alpaca api to call.  Valid values are "1" or "2".</param>
         public RestClient(
             String keyId,
             String secretKey,
             Uri alpacaRestApi,
             Uri polygonRestApi,
             Uri alpacaDataApi,
-            Boolean isStagingEnvironment)
+            Boolean isStagingEnvironment,
+            Int32 maxRetryAttempts,
+            IEnumerable<Int32> retryHttpStatuses,
+            String apiVersion)
         {
             keyId = keyId ?? throw new ArgumentException(nameof(keyId));
             secretKey = secretKey ?? throw new ArgumentException(nameof(secretKey));
+            if (maxRetryAttempts < 1) throw new ArgumentException(nameof(maxRetryAttempts));
+            _alpacaApiVersion = apiVersion ?? "1";
+            if (!_supportedApiVersions.Contains(apiVersion)) throw new ArgumentException(nameof(apiVersion));
```

Comment:
Logic error - if null value will be provided in apiVersion argument it will be replaced with correct fallback value 1 but on next line you will have exception. Most probably you want to use _alpacaApiVersion field in this Contains check.

Reply:
Thank you.  Nice catch.  Fixed.